tests/components/bsblan/test_climate.py

```diff
@@ -0,0 +1,186 @@
+"""Tests for the BSBLAN climate component."""
+
+from unittest.mock import AsyncMock, MagicMock
+
+from bsblan import BSBLANError, State
+import pytest
+
+from homeassistant.components.bsblan.climate import BSBLANClimate
+from homeassistant.components.bsblan.coordinator import BSBLanCoordinatorData
+from homeassistant.components.climate import PRESET_ECO, PRESET_NONE, HVACMode
+from homeassistant.const import UnitOfTemperature
+from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
+
+
+class MockStateValue:
+    """Mock class for a BSBLan State value."""
+
+    def __init__(self, value) -> None:
+        """Initialize the mock class."""
+        self.value = value
+
+
+@pytest.fixture
+def mock_state() -> MagicMock:
+    """Create a mock BSBLan State object."""
+    state = MagicMock(spec=State)
+    state.current_temperature = MockStateValue("22.5")
+    state.target_temperature = MockStateValue("21.0")
+    state.hvac_mode = MockStateValue("heat")
+    return state
+
+
+@pytest.fixture
+def coordinator_mock(mock_state) -> MagicMock:
+    """Create a mock BSBLan coordinator."""
+    coordinator = AsyncMock()
+    coordinator.data = BSBLanCoordinatorData(state=mock_state())
+    return coordinator
```

Comment:
Don't create classes in the tests. We want to set up the integration like it does in real life and then observe the state machine